form/create_custom_field_type.rst

```diff
@@ -14,93 +14,27 @@ Creating Form Types Based on Symfony Built-in Types
-By convention they are stored in the ``src/Form/Type/`` directory::
-
-    // src/Form/Type/ShippingType.php
-    namespace App\Form\Type;
-
-    use Symfony\Component\Form\AbstractType;
-    use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
-    use Symfony\Component\OptionsResolver\OptionsResolver;
-
-    class ShippingType extends AbstractType
-    {
-        public function configureOptions(OptionsResolver $resolver)
-        {
-            $resolver->setDefaults([
-                'choices' => [
-                    'Standard Shipping' => 'standard',
-                    'Expedited Shipping' => 'expedited',
-                    'Priority Shipping' => 'priority',
-                ],
-            ]);
-        }
-
-        public function getParent()
-        {
-            return ChoiceType::class;
-        }
-    }
-
-The ``configureOptions()`` method, which is explained later in this article,
-defines the options that can be configured for the form type and sets the
-default value of those options.
-
-The ``getParent()`` method defines which is the form type used as the base of
-this type. In this case, the type extends from ``ChoiceType`` to reuse all of
-the logic and rendering of that field type.
-
-.. note::
-
-    The PHP class extension mechanism and the Symfony form field extension
-    mechanism are not the same. The parent type returned in ``getParent()`` is
-    what Symfony uses to build and manage the field type. Making the PHP class
-    extend from ``AbstractType`` is only a convenience way of implementing the
-    required ``FormTypeInterface``.
-
-Now you can add this form type when :doc:`creating Symfony forms </forms>`::
-
-    // src/Form/Type/OrderType.php
-    namespace App\Form\Type;
-
-    use App\Form\Type\ShippingType;
-    use Symfony\Component\Form\AbstractType;
-    use Symfony\Component\Form\FormBuilderInterface;
-
-    class OrderType extends AbstractType
-    {
-        public function buildForm(FormBuilderInterface $builder, array $options)
-        {
-            $builder
-                // ...
-                ->add('shipping', ShippingType::class)
-            ;
-        }
-
-        // ...
-    }
-
-That's all. The ``shipping`` form field will be rendered correctly in any
-template because it reuses the templating logic defined by its parent type
-``ChoiceType``. If you prefer, you can also define a template for your custom
-types, as explained later in this article.
+You can read a dedicated article on this topic in
+":doc:`</form/create_custom_choice_type>`".
 
 Creating Form Types Created From Scratch
 ----------------------------------------
 
+Form types are PHP classes that implement :class:`Symfony\\Component\\Form\\FormTypeInterface`,
+but you should instead extend from :class:`Symfony\\Component\\Form\\AbstractType`,
+which already implements that interface and provides some utilities.
+By convention they are stored in the ``src/Form/Type/`` directory.
```

Comment:
Why tell this? Is this convention a requirement for anything to work?

Reply:
Again this is just moved from above, this not required but we do encourage to get organised since we can implement a lot of things (covered in other articles) such as type extensions, data transformers, data mappers, or listeners.